### PR TITLE
Add new rocksdb configurations

### LINKED ISSUE #6929: Backport non-memory related RocksDB configuration to 0.26
**Description**

We introduced optimized RocksDB configuration in 1.0 to soft cap the memory usage, ensure snapshots are not fragmented, and generally limit the the resource consumption of RocksDB (while keeping the performance on par).

While the solution relied on dropping the high number of column families - which would break backwards compatibility - we can at least back port part of the settings which are not related to having multiple column families. So all non-column family settings, and all non-memory related (no cache, no memtable) column family settings.

This includes not only the changes in the zb-db module but also the new `RocksDbCfg` usage in `BrokerCfg` (though removing any non-applicable settings).

NOTE: If the user specifies custom RocksDB config, then we should keep the old behavior and have that override any column family settings we might specify.

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Related issues

closes #6929 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Comment by deepthidevaki** [comment]
@npepinpe The PR is not yet done. But could you please check the new configurations `ZeebeRocksDbFactory`? Any params missing or should be removed?

File: zb-db/src/main/java/io/zeebe/db/impl/rocksdb/RocksDbConfiguration.java
Comment (npepinpe) [inline zb-db/src/main/java/io/zeebe/db/impl/rocksdb/RocksDbConfiguration.java:14]: I think we can have only one setting, that's fine. And just use it below, adapting the comment a little.

File: zb-db/src/main/java/io/zeebe/db/impl/rocksdb/ZeebeRocksDbFactory.java
Comment (npepinpe) [inline zb-db/src/main/java/io/zeebe/db/impl/rocksdb/ZeebeRocksDbFactory.java:134]: setCompactionPriority is added twice, here and on line 137

File: zb-db/src/main/java/io/zeebe/db/impl/rocksdb/ZeebeRocksDbFactory.java
Comment (npepinpe) [inline zb-db/src/main/java/io/zeebe/db/impl/rocksdb/ZeebeRocksDbFactory.java:204]: Would also add `setAvoidFlushDuringShutdown` which can also help prevent having too many small SST files (though I'm not sure by how much).

File: zb-db/src/main/java/io/zeebe/db/impl/rocksdb/ZeebeRocksDbFactory.java
Comment (npepinpe) [inline zb-db/src/main/java/io/zeebe/db/impl/rocksdb/ZeebeRocksDbFactory.java:237]: I don't see any issues with adding a bloom filter here - it will help with performance (theoretically) with little downsides, no? Let's say we have a 1% false positive rate, i.e. 10 bits for the filter, storing 1,000,000 items would take up ~1200KB, so I don't see much downsides. It's also unclear if it would help the performance at all, but from my pov it can't hurt

File: zb-db/src/main/java/io/zeebe/db/impl/rocksdb/ZeebeRocksDbFactory.java
Comment (npepinpe) [inline zb-db/src/main/java/io/zeebe/db/impl/rocksdb/ZeebeRocksDbFactory.java:139]: All the L0 settings are missing - I understand that we used to base the values of off the memtables, which we don't configure here, but we should still define them. This will help us have a bit more control on the compaction that gets triggered for L0->L1. Considering memtables by default are 64MB, we could just do it for every second memtable? We could omit the stop writes just to be safe.

```java
    // L-0 means immediately flushed memtables
        .setLevel0FileNumCompactionTrigger(2)
        .setLevel0SlowdownWritesTrigger(3)
```

**Review by npepinpe — COMMENTED** [review]
:rocket: 

Only minor suggestions

File: zb-db/src/main/java/io/zeebe/db/impl/rocksdb/ZeebeRocksDbFactory.java
Comment (deepthidevaki) [reply zb-db/src/main/java/io/zeebe/db/impl/rocksdb/ZeebeRocksDbFactory.java:244]: I guess we can add bloom filter configs also.

File: zb-db/src/main/java/io/zeebe/db/impl/rocksdb/ZeebeRocksDbFactory.java
Comment (deepthidevaki) [reply zb-db/src/main/java/io/zeebe/db/impl/rocksdb/ZeebeRocksDbFactory.java:144]: Was not sure about this. The default value for `Level0FileNumCompactionTrigger` is 4 which I thought is already small enough. So was not sure if there is any advantage in configuring it.

Comment (npepinpe) [reply zb-db/src/main/java/io/zeebe/db/impl/rocksdb/ZeebeRocksDbFactory.java:144]: You can leave it for now then - I'm also not 100% sure what it means concretely. I think a smaller value will trigger compaction of L0 to L1 faster, which I hope would help with fragmentation, but I'm also not sure. 4 * 64MB seems like a lot potentially (if we assume L0 files can be as big as a single memtable), but I'm also not 100% sure about it.

**Comment by deepthidevaki** [comment]
@npepinpe PR is ready to review. I have split it into 3 commits. The first two commits are only refactoring and renaming. 

A cluster is running on namespace `dd-0-26-rocksdb-config` and the base branch `dd-base-0-26` with low load. Once the PR is approved, I will also run a high load benchmark.

**Review by npepinpe — APPROVED** [review]
Looks good! Please update the YAML files in the dist module as well, as they act as templates for users. But no need for another review for this.

**Comment by deepthidevaki** [comment]
I have update the config templates for documentation. Meanwhile, also found that `disableWal` parameters was not applied.

**Comment by npepinpe** [comment]
I wouldn't expect them to limit - I was mostly hoping it would slow the fragmentation rate, but with 51 LSMs with varying traffic, it's unlikely that it will effectively limit it.

**Pull request merged**
